### ISSUE TITLE
Getting 404s on sourcemaps

### ISSUE THREAD

**Issue by joshbedo** [gobblejs/gobble#67]
Recently with our builds we've been getting 404s on all of our sourcemaps. I tried clearing out .gobble, .gobble-build and also node_modules/.bin/gobble but haven't had any luck solving the problem. I saw another issue recently posted about sourcemaps and was wondering if it was related? Looking for any kind of help since we can't debug right now :)

![](https://api.monosnap.com/rpc/file/download?id=1WfdPpHjYJhJ29XF10gxxgkdqqNK6n)

**Comment by Rich-Harris** [comment]
Sorry about this - yes, I'm fairly sure it's related. I just merged #66 which ought to fix it, I'm just tidying up a couple of other bits and pieces then I'll push a new version out, hopefully by the end of the day.

**Comment by Rich-Harris** [comment]
@joshbedo I just released 0.10.0, which should fix this issue. Let me know if not, thanks

**Comment by Rich-Harris** [comment]
Oh, and you'll probably want to `npm install -g gobble-cli@latest` while you're at it, otherwise some of the logging will look borked

**Comment by joshbedo** [comment]
So i just tried to upgrade and i'm having issues with `peerDependency` requirements. It says i need `0.7.1` for `gobble-grunt`. Any ideas or suggestions? Thanks!

![](https://d1ro8r1rbfn3jf.cloudfront.net/ms_65558/v3W6JtmHHUex4tcY0JJQquuYK7eCFB/1.%2Btmux%2B%2528tmux%2529%2B2015-05-06%2B12-47-43.png?Expires=1431017407&Signature=cmqDER1mGJsLIy7gD2V5bhsMI3EEfkVREhjzDShbnyX8YcWZF4t-UjfNChxcBSGOyOsCCE242AhBBqedY4lpBslUz-5eVe-NmC~5FEZ9KgpEFn2BOwlKzjiKq~rU~45V4baYI-qrLQQNm94l2BzTz4szNNnwadP1cuOYZnSouQfZ98Ha4D2yw-D2UAx9QXLjr2o3H-BP-LxdZMzcIxqVNnVv28op5THZ7xy7xplJqYZs-qHKzZFT2J02wCAjIHY8~K9HvIKEpKzDPSJDGDD1PDwldS1CtvUiwAvoN-uDSROa3uABJ5xBL4SLKTV3rbZ7WNv~YJ3xSCK3miMm33qF2A__&Key-Pair-Id=APKAJHEJJBIZWFB73RSA)

**Comment by Rich-Harris** [comment]
Ack, I struggle with npm's error messages. Have to dash out of the office right now so can't investigate, but does [this article](https://futurestud.io/blog/how-to-fix-npm-package-does-not-satisfy-its-siblings-peerdependencies-requirements/) help any?

**Comment by evs-chris** [comment]
You can manually edit the gobble-grunt package.json to have `'gobble": "^0"` as a quick workaround.

**Comment by joshbedo** [comment]
Hey @evs-chris i tried what you suggested and still get the same error.

**Comment by evs-chris** [comment]
That's odd... is there more than one grunt-gobble in your node_modules tree? `find . -name grunt-gobble` or something like that from your project root should turn up any instances of grunt-gobble, and editing their package.json files *should* skirt the error. I wouldn't expect there to be more than one. What npm command are you running that's throwing the error?

**Comment by joshbedo** [comment]
So we only have one instance of `grunt-gobble` i modified its `package.json` to use `^0` and that seems to have fixed the npm install error. Now when building i'm getting this error.

![](https://api.monosnap.com/rpc/file/download?id=zJzvYg2ySdcreTkIawdpZCJCzA4rOM)

**Comment by evs-chris** [comment]
That sourcemap transform error should not be fatal. Does your project still serve correctly?

**Comment by joshbedo** [comment]
Yes it does, its just throwing the error on vendors on a few other areas.. not sure why or how to fix.. i've never seen an EMFILE error.

**Comment by joshbedo** [comment]
Although now i'm noticing sometimes it stops the build from finishing.

**Comment by Rich-Harris** [comment]
Whoops, that EMFILE is probably happening because the update version of [sander](https://github.com/rich-harris/sander) was using fs instead of [graceful-fs](https://github.com/isaacs/node-graceful-fs). I've just fixed that and released a new version. If you remove `gobble` from `node_modules` and reinstall it, it'll get the latest sander and you should no longer see that error.

Also, if you `npm install -g gobble-cli@0.4.2` you'll get slightly nicer logging

**Comment by joshbedo** [comment]
So i'm still getting a similar error after upgrading :\ 

![](https://api.monosnap.com/rpc/file/download?id=0WAkCt9tX3Bsxq4czoh4TtteQrDzKE)

**Comment by Rich-Harris** [comment]
Is the build failing as a result? ENOENTs should be harmless (they do need better logging admittedly, it looks positively alarming when you see them like that), in theory it'll just skip those files

**Comment by aubergene** [comment]
I'm getting this message too, the build is fine, just the sourcemap isn't generated. gobble-cli version 0.4.2

```
gobble: server listening on port 4567
gobble: livereload server running
gobble: 01-sass done in 56ms
gobble: 02-merge done in 5ms
∙∙∙∙∙∙∙ processing sourcemaps
gobble: ENOENT, open '/Users/aubergene/code/tmp/gobble-test/.gobble/main.css.map'
>>>
Error: ENOENT, open '/Users/aubergene/code/tmp/gobble-test/.gobble/main.css.map'
    at Error (native)
<<<

gobble: processed sourcemaps in 14ms
gobble: build completed in 83ms. Listening for changes...
```

**Comment by Rich-Harris** [comment]
@aubergene can you share your gobblefile (or at least the bit with sass and the subsequent merge)? That's a weird place for it to be looking for the sourcemap, which makes me wonder if it's getting confused with relative paths or something.

Aaaand now that I look at @joshbedo's screenshot again, it looks like it's exactly the same problem. Are you using sass as well Josh?

**Comment by Rich-Harris** [comment]
Ah, I've found the problem. gobble-sass was generating a sourcemap without understanding where it was writing the actual CSS file to, so the relative path was getting messed up. Not a problem with gobble itself (though it was a recent update to gobble that exposed the problem) - will have a fix released shortly

**Comment by Rich-Harris** [comment]
I've just released gobble-sass@0.4.1, which ensures the relative sourcemap path is relative to the right thing. I also updated gobble itself, which contains a couple of small fixes related to CSS sourcemap URLs.

**Comment by joshbedo** [comment]
Hmm it seems like after upgrading im still getting the same error.

![](https://api.monosnap.com/rpc/file/download?id=QHoJOt19xZbuA5f5fnxToF2MrCBoI3)

**Comment by joshbedo** [comment]
Not as many errors though, only three this time. One looks really weird, i think its because of bower components. 

![](https://api.monosnap.com/rpc/file/download?id=rVTj421SmA6PkxYY7K8pD8cYBQtBSN)

**Comment by Rich-Harris** [comment]
Those *are* weird errors - the 'cannot call method 'trace' of undefined', 'unexpected end of input' and ENOENT on `jquery/)(.*)` are particularly concerning. I don't suppose you're in a position to share the repo, or a stripped-down version that reproduces those errors?

**Comment by joshbedo** [comment]
Its a really large repo, i can share the structure and bower/package.json file? What would be the most helpful?

**Comment by Rich-Harris** [comment]
If you can put `gobblefile.js`, `[bower|package].json`, and a bare-bones `app` folder, maybe with a couple of `.js`, `.coffee` and `.es6.js` files in `app/assets`, that ought to do the trick. Thanks!

**Comment by aubergene** [comment]
I updated gobble and gobble-sass and it ran without errors, thanks! However the sourcemap is generated with the wrong path, so it doesn't work in the browser, the output was `"sources":["../.gobble-build/02-merge/1/main.scss"]`.

**Comment by Rich-Harris** [comment]
@aubergene There's an open issue for that - #69 - basically, as far as node-sass is concerned, the source file *is* `02-merge/1/main.scss`, and we need to jump through some hoops to trace it back to the actual file. It should still work though, because the `sourcesContent` is populated:

![screen shot 2015-05-08 at 2 53 20 pm](https://cloud.githubusercontent.com/assets/1162160/7543423/35ae3ca8-f592-11e4-9b37-9892a181d87e.png)

![screen shot 2015-05-08 at 2 55 13 pm](https://cloud.githubusercontent.com/assets/1162160/7543436/43fb05c0-f592-11e4-9f8f-0fb55529d189.png)

Are you not seeing the `main.scss` in devtools?

**Comment by aubergene** [comment]
I only see `main.css`, even though `sourcesContent` is in the sourcemap and looks correct. Here's the example repo I've been playing with https://github.com/aubergene/gobble-example

**Comment by Rich-Harris** [comment]
Do you have CSS sourcemaps enabled in devtools? Your repo works for me... You need to check this box:

![screen shot 2015-05-11 at 9 33 01 am](https://cloud.githubusercontent.com/assets/1162160/7565791/d14b6d72-f7c0-11e4-8c90-d08d5f253b50.png)

Also, I released a new version of gobble-sass which upgrades to the just-released node-sass v3.0.0

**Comment by aubergene** [comment]
Yep, it's enabled. It works fine on both my work machines, so it's probably something odd with my settings on laptop. I think it's fine to close this ticket. Thanks!

**Comment by Rich-Harris** [comment]
> It works fine on both my work machines

computers ¯\ _ (ツ) _ /¯

Will leave this open for the time being, for @joshbedo

**Comment by joshbedo** [comment]
Sorry i havent had a chance to post the files, i'll try to post something tonight.
